rs/nns/governance/api/src/subnet_rental.rs

```diff
@@ -0,0 +1,40 @@
+use ic_base_types::PrincipalId;
+use std::str::FromStr;
+
+/// A proposal payload for a subnet rental request,
+/// used to deserialize `ExecuteNnsFunction.payload`,
+/// where `ExecuteNnsFunction.nns_function == NnsFunction::SubnetRentalRequest as i32`.
+/// Also used to serialize the subnet rental request payload in `ic-admin`.
+#[derive(candid::CandidType, candid::Deserialize, serde::Serialize, Clone, Debug)]
+pub struct SubnetRentalRequest {
```

Comment:
I think the same thing apply here similar to BitcoinSetConfig - it seems to me that this should be internal to NNS Governance.

Reply:
Do you think I should just duplicate the types in registry admin and private to NNS governance?

It seems like they're both instances of the same thing.

Reply:
Perhaps we can indeed think of every type that can represent the `ExecuteNnsFunction::payload` blob as part of the NNS Governance API. In that case, those seem to be in the right crate. IMHO it might make sense to put them into the same file, and clearly define what the file is for, although I don't have a strong opinion on that.